tests/kernel/testNumberTraits.cpp

```diff
@@ -0,0 +1,203 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file
+ * @author Roland Denis (\c denis@math.univ-lyon1.fr )
+ * CNRS, Institut Camille Jordan (UMR 5208), Université Lyon 1, France
+ *
+ * @date 2019/02/21
+ *
+ * This file is part of the DGtal library.
+ */
+
+#include <limits>
+
+#include "DGtal/base/BasicTypes.h"
+#include "DGtal/kernel/NumberTraits.h"
+
+#include "DGtalCatch.h"
+
+/// Transform a std::integral_constant<bool, value> to the corresponding DGtal tag.
+template <typename T>
+struct IntegralCstToTag
+{
+  using type = DGtal::TagUnknown;
+};
+
+template <>
+struct IntegralCstToTag<std::true_type>
+{
+  using type = DGtal::TagTrue;
+};
+
+template <>
+struct IntegralCstToTag<std::false_type>
+{
+  using type = DGtal::TagFalse;
+};
+
+/// Transform a boolean value to the corresponding DGtal tag.
+template <bool V>
+struct ValueToTag
+{
+  using type = DGtal::TagUnknown;
+};
+
+template <>
+struct ValueToTag<true>
+{
+  using type = DGtal::TagTrue;
+};
+
+template <>
+struct ValueToTag<false>
+{
+  using type = DGtal::TagFalse;
+};
+
+#define REQUIRE_SAME_TAG(a, b)    REQUIRE( (std::is_same<a, typename IntegralCstToTag<b>::type>::value) )
```

Comment:
nice macros;)

Reply:
:sweat_smile:

Reply:
I realize that I don't even use this macro...

Reply:
^^